packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -67,6 +67,8 @@ class Context {
     return file.existsSync();
   }
 
+  Directory directoryFromPath(String path) => Directory(path);
```

Comment:
If you want to go the extra mile, it might make sense to introduce a `FileSystem` type that's passed in to the `Context` through its constructor. The test would extend that `FileSystem` type & override stuffs it wants to mock. The current setup is a bit gross.

Feel free to punt this to another pull request (or never 😂). What you're doing is perfectly fine and follows the existing structure